bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -105,49 +119,64 @@ pub mod pallet {
 		/// Convert versioned location failure
 		UnsupportedLocationVersion,
 		/// Check location failure, should start from the dispatch origin as owner
-		OwnerCheck,
+		InvalidAssetOwner,
 		/// Send xcm message failure
 		Send,
 		/// Withdraw fee asset failure
 		FundsUnavailable,
+		/// Convert to reanchored location failure
+		LocationConversionFailed,
 	}
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		/// Call `create_agent` to instantiate a new agent contract representing `origin`.
-		/// - `origin`: Can be any sovereign `Location`
 		/// - `fee`: Fee in Ether paying for the execution cost on Ethreum
 		#[pallet::call_index(1)]
 		#[pallet::weight(T::WeightInfo::create_agent())]
 		pub fn create_agent(origin: OriginFor<T>, fee: u128) -> DispatchResult {
 			let origin_location = T::CreateAgentOrigin::ensure_origin(origin)?;
 
-			Self::burn_fees(origin_location.clone(), fee)?;
+			// Burn Ether Fee for the cost on ethereum
+			Self::burn_for_teleport(&origin_location, &(T::FeeAsset::get(), fee).into())?;
+
+			// Burn RemoteExecutionFee for the cost on bridge hub
+			Self::burn_for_teleport(&origin_location, &T::RemoteExecutionFee::get())?;
 
-			let call = SnowbridgeControl::Control(ControlCall::CreateAgent {
-				location: Box::new(VersionedLocation::from(origin_location.clone())),
+			let reanchored_location = origin_location
+				.clone()
+				.reanchored(&T::BridgeHubLocation::get(), &T::UniversalLocation::get())
+				.map_err(|_| Error::<T>::LocationConversionFailed)?;
+
+			let call = BridgeHubRuntime::EthereumSystem(EthereumSystemCall::CreateAgent {
+				location: Box::new(VersionedLocation::from(reanchored_location.clone())),
 				fee,
 			});
 
+			let topic = blake2_256(&("SnowbridgeFrontend", call.clone()).encode());
+
 			let xcm: Xcm<()> = vec![
-				ReceiveTeleportedAsset(T::DeliveryFee::get().into()),
-				PayFees { asset: T::DeliveryFee::get() },
+				// Burn some DOT fees from the origin on AH and teleport to BH which pays for
+				// the execution of Transacts on BH.
+				ReceiveTeleportedAsset(T::RemoteExecutionFee::get().into()),
+				PayFees { asset: T::RemoteExecutionFee::get() },
 				Transact {
 					origin_kind: OriginKind::Xcm,
 					call: call.encode().into(),
 					fallback_max_weight: None,
 				},
+				ExpectTransactStatus(MaybeErrorCode::Success),
+				SetTopic(topic),
```

Comment:
We should not let the leftover execution fee be trapped. The origin user won't be able to easily claim it on BH.

Easiest to just transfer the leftover fee to the sovereign account of the `/Para(AssetHub)/PalletIndex(SystemFrontend)`

```suggestion
				DepositAsset {
					assets: Wild(AllOf { id: message.execution_fee.id, fun: WildFungible }),
					beneficiary: SELF,
				}
				SetTopic(topic),
```

And note that `ExpectTransactStatus` will cause the xcm execution to fail before reaching `DepositAsset`.

Reply:
Since we use `Unpaid` now, there is no need to `DepositAsset`